flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -403,6 +403,41 @@ class CodeGenerator(
          |  }""".stripMargin
     }
 
+    def generateMergeTwoRows(
+        accTypes: Array[String],
+        aggs: Array[String]): String = {
+
+      val sig: String =
+        j"""
+           |  public org.apache.flink.types.Row mergeTwoRows(
+           |    org.apache.flink.types.Row a,
+           |    org.apache.flink.types.Row b)
+           |    """.stripMargin
+      val merge: String = {
+        for (i <- aggs.indices) yield
+          j"""
+             |    ${accTypes(i)} aAcc$i = (${accTypes(i)}) a.getField($i);
+             |    ${accTypes(i)} bAcc$i = (${accTypes(i)}) b.getField($i);
+             |    java.util.ArrayList<${accTypes(i)}> accumulators$i
```

Comment:
I was planning to remove `trait Accumulator` very soon, we cannot reuse the list after that.